packages/x-data-grid/src/hooks/features/pagination/gridPaginationUtils.ts

```diff
@@ -1,27 +1,22 @@
 import { DataGridProcessedProps } from '../../../models/props/DataGridProps';
-import { buildWarning } from '../../../utils/warning';
 import { GridSignature } from '../../utils';
 
 const MAX_PAGE_SIZE = 100;
 
 export const defaultPageSize = (autoPageSize: boolean) => (autoPageSize ? 0 : 100);
 
-export const getPageCount = (rowCount: number, pageSize: number): number => {
+export const getPageCount = (rowCount: number, pageSize: number, page: number): number => {
   if (pageSize > 0 && rowCount > 0) {
     return Math.ceil(rowCount / pageSize);
   }
 
+  if (rowCount === -1) {
+    return page + 1;
```

Comment:
Since `page` is 0-based, should we assume there's always one more page after the current one when `rowCount === -1`?

```suggestion
    return page + 1 + 1;
```